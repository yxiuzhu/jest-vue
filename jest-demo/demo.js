import Axios from "axios";

export const runCallback = (callback) => {
  callback();
};

export const createObject = (classItem) => {
  new classItem();
};

export const getData = () => {
  return Axios.get("/api").then((res) => res.data);
};

export const generateConfig = () => {
  return {
    server: "http://localhost",
    time: new Date(),
  };
};
