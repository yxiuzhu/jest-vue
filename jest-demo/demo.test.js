import axios from "axios";
import { createObject, runCallback, getData, generateConfig } from "./demo";
jest.mock("axios");

test("执行回调函数1", () => {
  const func = jest.fn(); // 生成mock函数
  runCallback(func); // 执行mock函数
  expect(func).toBeCalled(); // 查看是否已经执行mock函数
});

test("执行回调函数2", () => {
  const func = jest.fn(() => {
    return "456";
  });
  runCallback(func);
  expect(func.mock.calls.length).toBe(1);
});

test("执行回调函数3", () => {
  const func = jest.fn();
  func.mockReturnValueOnce("freeman");
  runCallback(func);
  expect(func.mock.results[0].value).toBe("freeman");
});

test.only("执行异步函数", async () => {
  axios.get.mockResolvedValue({ data: "success" });
  await getData().then((data) => {
    expect(data).toBe("success");
  });
});

test.only("测试配置文件", () => {
  expect(generateConfig()).toMatchSnapshot({
    time: expect.any(Date),
  });
});
