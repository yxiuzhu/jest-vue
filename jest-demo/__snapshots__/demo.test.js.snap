// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试配置文件 1`] = `
Object {
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
